File: in_Java/1282 Group the People Given the Group Size They Belong To.java
class Solution {
    public List<List<Integer>> groupThePeople(int[] groupSizes) {
        List<List<Integer>> res = new ArrayList<>();
        if(groupSizes == null || groupSizes.length == 0) return res;
        HashMap<Integer,List<Integer>> hm = new HashMap<>();
        List<Integer> cur;
        for(int i = 0; i < groupSizes.length; i++){
            if(!hm.containsKey(groupSizes[i])){
                cur = new ArrayList<Integer>();
                cur.add(i);
                hm.put(groupSizes[i], cur);
            }else{
                cur = hm.get(groupSizes[i]);
                cur.add(i);
                hm.put(groupSizes[i], cur);
            }
        }
        for(int k : hm.keySet()){
            cur = hm.get(k);
            int len = k;
            while(cur.size() != 0){
                List<Integer> temp = new ArrayList<>();
                while(len-- != 0){                    
                    temp.add(cur.remove(0));
                }
                res.add(temp);
                len = k;
            }            
        }
        return res;
    }
}

File: in_Python/1282 Group the People Given the Group Size They Belong To.py
class Solution(object):
    def groupThePeople(self, groupSizes):
        """
        :type groupSizes: List[int]
        :rtype: List[List[int]]
        """
        n = len(groupSizes)
        dict = {}
        for gi in range(n):
            key = groupSizes[gi]
            if key not in dict:
                temp = []
                temp.append(gi)
                dict[key] = temp
            else:
                temp = dict[key]
                temp.append(gi)
                dict[key] = temp
        res = []
        for key in dict:
            members = dict[key]
            num = len(dict[key]) / key
            while num != 0:
                temp = []
                num -= 1
                for j in range(key):
                    temp.append(members.pop())
                res.append(temp)
        return res
                
                